src/sentry/auth/helper.py

```diff
@@ -226,22 +228,24 @@ def handle_attach_identity(
         self,
         identity: Identity,
         member: Optional[OrganizationMember] = None,
+        user: Optional[User] = None,
```

Comment:
why do you have to take a user as a parameter here? why can't you use `self.user`?

Reply:
when we're trying to authenticate the user, self.user is AnonymousUser which isn't the user that needs to be authenticated, so i needed to pass in the user in order to create an auth identity

Reply:
why can't you use `self._get_user(identity)` then? (also, generally, the user should be defined on the request once testing in the UI, but its fine for it to work without). 

I would also be in favor of renaming `self.user` on this class to `self.request_user`, and adding `self.acting_user` as a new property